.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimeParser.cs

```diff
@@ -249,8 +249,25 @@ private DateTimeResolutionResult MergeDateAndTime(string text, DateObject refere
                 ret.Comment = Constants.Comment_AmPm;
             }
 
-            ret.FutureValue = DateObject.MinValue.SafeCreateFromValue(futureDate.Year, futureDate.Month, futureDate.Day, hour, min, sec);
-            ret.PastValue = DateObject.MinValue.SafeCreateFromValue(pastDate.Year, pastDate.Month, pastDate.Day, hour, min, sec);
+            // Handle case like "Wed Oct 26 15:50:06 2016" which year and month separated by time.
```

Comment:
Can we split this part to a separate function? That will help readability.

Reply:
Changed.

Reply:
Didn't see the change..

Reply:
> Didn't see the change..

Do you mean I should write a function? I think this one can be here like the "Handle morning, afternoon".

Reply:
Yes, since this function already contains a large body of code, we should make the function as neat as possible if we can.

Reply:
I have to use up to 8 local variables here to change the value of ret. Are you sure about this?

Reply:
Yes, it may need further refactor, but it's fine with this PR.